test/attr/open_objc.swift

```diff
@@ -11,8 +11,14 @@ open class InvalidOpenExtension {
 extension InvalidOpenExtension {
   open func nonObjcOpenMethod() { } // expected-warning {{non-'@objc' methods in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
   open var nonObjcOpenVar: Int { 3 }  // expected-warning {{non-'@objc' properties in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
+  open subscript(_ index: Int) -> Int { 3 } // expected-warning {{non-'@objc' properties in extensions cannot be overridden; use 'public' instead}} {{3-7=public}}
```

Comment:
Sweet!

I think this just needs a minor reworking so that the subscript is not referred to as a property; there are other diagnostics that have to deal with this so it'll be straightforward to adapt to your use case.